### PR TITLE
TINY-6309: Caret moves to first cell when deleting last row in colgroup table

### PR DESCRIPTION
**Pull request by danoaky-tiny**
Related Ticket: https://ephocks.atlassian.net/browse/TINY-6309

Description of Changes:
Caret/cursor now correctly moves to the directly adjacent cell when a row or column is deleted.
The Jira ticket states that both row and column deletion are affected. But as of right now, only deleting the last row within a table that has a colgroup still has this bug. So this change only puts in a fix for row deletions.
I've also put in tests for column deletion as well.

Pre-checks:
* [x] Changelog entry added
* [X] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):
#5952

### CODE REVIEW

File: versions.txt [hunk 1]
```diff
@@ -7,3 +7,4 @@ mcagar@9.0.0
 alloy@14.0.0
 bridge@4.7.0
 sugar@9.3.0
+snooker@11.1.1
```
Comment (spocke) [inline versions.txt:10]: Does this need to be a minor since this is a bug fix only it could go into a patch.

**Review by spocke — APPROVED** [review]

File: versions.txt (hunk 1 above)
Comment (danoaky-tiny) [reply versions.txt:10]: Oh right, is the current version of snooker not already 11.1.0?
https://github.com/tinymce/tinymce/blob/693fef6287304759bc2f3dc31f76e9e74a68d8d1/modules/snooker/package.json#L4
Or am I misunderstanding?

File: modules/tinymce/src/models/dom/test/ts/browser/table/command/TableDeleteRowTest.ts [hunk 2]
```diff
@@ -80,4 +80,43 @@ describe('browser.tinymce.models.dom.table.command.TableDeleteRowTest', () => {
...
+        const originalTBody = tr(4, 3);
+        editor.setContent(createTable(originalTBody.join('')));
+
+        // Delete last row:
+        TinySelections.setCursor(editor, [ ...pathToBody, 3, 1, 0 ], 0);
+        editor.execCommand('mceTableDeleteRow');
+        TinyAssertions.assertCursor(editor, [ ...pathToBody, 2, 1, 0 ], 3);
+        TinyAssertions.assertContent(editor, createTable(originalTBody.slice(0, -1).join('')));
+
+        // Delete first row:
```
Comment (ltrouton) [inline modules/tinymce/src/models/dom/test/ts/browser/table/command/TableDeleteRowTest.ts:112]: I would probably have deleting the first row and deleting the last row as separate `it` block test cases

File: modules/tinymce/src/models/dom/test/ts/browser/table/command/TableDeleteRowTest.ts [hunk 3]
```diff
@@ -80,4 +80,43 @@ describe('browser.tinymce.models.dom.table.command.TableDeleteRowTest', () => {
...
+  Arr.each([
+    {
+      label: 'no colgroup',
+      pathToBody: [ 0, 0 ],
+      createTable: (tbody: string) => `<table><tbody>${tbody}</tbody></table>`,
+    },
+    {
+      label: 'with colgroup',
+      pathToBody: [ 0, 1 ],
+      createTable: (tbody: string) => `<table><colgroup><col><col><col></colgroup><tbody>${tbody}</tbody></table>`,
```
Comment (ltrouton) [inline modules/tinymce/src/models/dom/test/ts/browser/table/command/TableDeleteRowTest.ts:97]: Just have to be mindful that the number of `col` elements is fixed here and that could be incorrect from changing `const originalTBody = tr(4, 3);`

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/models/dom/test/ts/browser/table/command/TableDeleteRowTest.ts (hunk 2 above)
Comment (danoaky-tiny) [reply modules/tinymce/src/models/dom/test/ts/browser/table/command/TableDeleteRowTest.ts:112]: I've now split them into separate `it` blocks

File: modules/tinymce/src/models/dom/test/ts/browser/table/command/TableDeleteRowTest.ts (hunk 3 above)
Comment (danoaky-tiny) [reply modules/tinymce/src/models/dom/test/ts/browser/table/command/TableDeleteRowTest.ts:97]: Good point, I've now refactored it to account for this

**Comment by spocke** [comment]
So this seems to be for 7.0

- [ ] Merge this after we make the 6.8 release

**Review by TheSpyder — APPROVED** [review]

File: versions.txt (hunk 1 above)
Comment (danoaky-tiny) [reply versions.txt]: Bumped snooker version to 11.2.1 now after the 6.8 release

**Review by mkzhx — APPROVED** [review]

**Review by MitchC1999 — APPROVED** [review]

**Pull request merged**